MapboxDirections/MBDirectionsResult.swift

```diff
@@ -0,0 +1,171 @@
+import Polyline
+
+/**
+ A `DirectionsResult` represents a result returned from either the Directions API or the Map Matching API.
```

Comment:
> A `DirectionsResult` represents a result returned from either the Mapbox Directions service.
> 
> You do not create instances of this class directly. Instead, you receive `Route` or `Match` objects when you request directions using the `Directions.calculate(_:completionHandler:)` or `Directions.calculateRoutes(matching:completionHandler:)` method.

“Mapbox Directions service” encompasses the Directions API and Map Matching API. At least that’s how it’s presented in the sidebar of the [API Documentation](https://www.mapbox.com/api-documentation/#directions) site.